padl/transforms.py

```diff
@@ -639,12 +641,15 @@ def pd_forward_device_check(self) -> bool:
         All transforms in forward need to be in same device as specified for
         the whole Pipeline.
         """
-        for layer in self.pd_forward.pd_layers:
+        pd_device = self.pd_device
+        pd_layers = self.pd_forward.pd_layers
```

Comment:
I don't think that's necessary, `self.pd_device` is being cached and pd_layers is only used once below.

Reply:
I don't think `self.pd_device` is being cached.

Reply:
I agree with self.pd_forward_layers

Reply:
Eh sorry not cached, but it's just a simple lookup, right? and this isn't called too often, so I'm not sure this warrants the slightly diminished readability. But either way is fine.